src/coreclr/jit/importer.cpp

```diff
@@ -18912,15 +18912,15 @@ void Compiler::impMakeDiscretionaryInlineObservations(InlineInfo* pInlineInfo, I
         }
     }
 
-    // Check if callee is a generic method when its caller is not
-    bool calleeIsGeneric = (info.compMethodInfo->args.callConv & CORINFO_CALLCONV_GENERIC) ||
-                           (info.compClassAttr & CORINFO_FLG_SHAREDINST);
-    bool callerIsGeneric = (rootCompiler->info.compMethodInfo->args.callConv & CORINFO_CALLCONV_GENERIC) ||
-                           (rootCompiler->info.compClassAttr & CORINFO_FLG_SHAREDINST);
-
-    if (!callerIsGeneric && calleeIsGeneric)
+    if ((info.compMethodInfo->args.callConv & CORINFO_CALLCONV_GENERIC) ||
+        (info.compClassAttr & CORINFO_FLG_SHAREDINST))
+    {
+        inlineResult->Note(InlineObservation::CALLEE_GENERIC);
```

Comment:
Maybe be more explicit that it's the sharing that is interesting?
```suggestion
        inlineResult->Note(InlineObservation::CALLEE_SHARED_GENERIC);
```

Reply:
https://github.com/dotnet/runtime/pull/53670#discussion_r647713036